### PR TITLE
feat(device): allow custom config variables and labels during onboarding

### PR DESCRIPTION
**Pull request by rrkumarshikhar**
… of creating the device

### CODE REVIEW

**Review by amitsingh21 — APPROVED** [review]

**Review by pallabpain — CHANGES_REQUESTED** [review]
Change to the scope to feat: 

@rrkumarshikhar

**Comment by rrkumarshikhar** [comment]
> Change to the scope to feat:
> 
> @rrkumarshikhar
updated the title

**Comment by pallabpain** [comment]
Remove the emoji if you intend to squash and merge. It will fail the semantic release workflow.

File: rapyuta_io/clients/device.py [hunk 1]
```diff
@@ -320,6 +323,9 @@ def _serialize(self):
                       '_ros_workspace']:
             if getattr(self, field):
                 device['config_variables'][field[1:]] = getattr(self, field)
+        for key, value in self.config_variables.items():
+            device['config_variables'][key] = value
+        device['labels'] = self.labels
```
Comment (ankitrgadiya) [inline rapyuta_io/clients/device.py:328]: question: Why did we have to loop for config_variables but for labels we assigned directly.

File: rapyuta_io/clients/device.py [hunk 2]
```diff
@@ -272,6 +273,8 @@ def __init__(self, name, runtime=None, runtime_docker=False, runtime_preinstalle
         self._ros_workspace = ros_workspace
         self.description = description
         self.python_version = python_version
+        self.config_variables = config_variables if config_variables else {}
+        self.labels = labels if labels else {}
```
Comment (ankitrgadiya) [inline rapyuta_io/clients/device.py:277]: suggestion: I think default None value makes the if check redundant, we can directly assign the values here without doing the if check.

Comment (pallabpain) [reply rapyuta_io/clients/device.py:277]: or you can do

labels = labels or {}

**Review by ankitrgadiya — APPROVED** [review]
Looks good.

**Review by ankitrgadiya — APPROVED** [review]

**Review by pallabpain — APPROVED** [review]

**Pull request merged**